cmd/preview/list_test.go

```diff
@@ -0,0 +1,103 @@
+package preview
+
+import (
+	"context"
+	"fmt"
+	"github.com/okteto/okteto/internal/test/client"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/okteto/okteto/pkg/types"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func Test_listPreview(t *testing.T) {
+	ctx := context.Background()
+	opts := ListFlags{}
+	var tests = []struct {
+		name     string
+		previews []types.Preview
+		err      error
+	}{
+		{
+			name: "List all previews",
+			previews: []types.Preview{
+				{
+					ID: "test",
+				},
+				{
+					ID: "test-1",
+				},
+			},
+		},
+		{
+			name:     "error retrieving preview list",
+			previews: nil,
+			err:      fmt.Errorf("error retrieving previews"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			okteto.CurrentStore = &okteto.OktetoContextStore{
+				Contexts: map[string]*okteto.OktetoContext{
+					"test": {
+						Name:     "test",
+						Token:    "test",
+						IsOkteto: true,
+						UserID:   "1",
+					},
+				},
+				CurrentContext: "test",
+			}
+			usr := &types.User{
+				Token: "test",
+			}
+			fakeOktetoClient := &client.FakeOktetoClient{
+				Preview: client.NewFakePreviewClient(&client.FakePreviewResponse{PreviewList: tt.previews, ErrList: tt.err}),
+				Users:   client.NewFakeUsersClient(usr),
+			}
+			_, err := getPreviewOutput(ctx, opts, fakeOktetoClient)
```

Comment:
we need to check the result too. You can use `assert.ElementsMatch(t, result, tt.expectedResult)`